File: Python-tests/Python-micropython/benchmark.py
#!/usr/bin/micropython
# -*- coding: utf8 -*-

#from __future__ import print_function

import fft
import utime

LOG2FFTSIZE = 12
FFT_REPEAT = 100

SIZE = 1<<LOG2FFTSIZE
xy = [0+0j]*SIZE

if __name__ == "__main__":

    for i in range(int(SIZE/2)):
        xy[i]= 1.

    for i in range(int(SIZE/2), SIZE):
        xy[i]=-1.

    timestart = utime.time()
    for i in range(FFT_REPEAT):
        fft_out = fft.fft(LOG2FFTSIZE, xy)
    eltime = 1000*(utime.time() - timestart)
    print ("%6d piece of %d pt FFT;  %9.5f ms/piece\n"%(FFT_REPEAT, 1<<LOG2FFTSIZE, eltime/FFT_REPEAT))

    for i in range(6):
        print (i, fft_out[i])

File: Nim/benchmark.nim
import fft    # here is my functions
import bitops
import std/[complex, strformat, times]

const LOG2FFTSIZE = 12
const FFT_REPEAT = 1000

const SIZE = cast[int](rotateLeftBits(1'u32, LOG2FFTSIZE))
const SIZE_PER2 = cast[int](rotateLeftBits(1'u32, LOG2FFTSIZE-1))

proc main() =
    var xy: array[SIZE, Complex32]
    for i in 0..<SIZE_PER2:
        xy[i] = complex(1.0'f32, 0.0'f32)
        xy[i+SIZE_PER2] = complex(-1.0'f32, 0.0'f32)

    var f = newfft()
    var fft_out: seq[Complex32];
    var timestart = epochTime()
    for i in 0..<FFT_REPEAT:
        fft_out = f.fft(LOG2FFTSIZE, xy)
    var eltime = 1000*(epochTime() - timestart)
    var s = fmt("{FFT_REPEAT:6} piece of {SIZE} pt FFT;  {eltime/FFT_REPEAT:9.5} ms/piece\n")
    echo s

    echo "bin        real             imag           absval";
    for i in 0..<6:
        var s = fmt("{i:3} {fft_out[i].re:16.4} {fft_out[i].im:16.4} {abs(fft_out[i]):16.4}")
        echo s

main()
